tests/unit/onyxCacheTest.js

```diff
@@ -0,0 +1,680 @@
+                        expect(cache.hasPendingTask('mockTask')).toBe(false);
+                    });
+            });
+        });
+
+        describe('getTaskPromise', () => {
+            it('Should return undefined when there is no stored value', () => {
+                // GIVEN empty cache with no started tasks
+
+                // WHEN a task is retrieved
+                const task = cache.getTaskPromise('mockTask');
+
+                // THEN it should be undefined
+                expect(task).not.toBeDefined();
+            });
+
+            it('Should return captured task when it exists', () => {
+                // GIVEN empty cache with no started tasks
+                // WHEN a unique task is started
+                const promise = Promise.resolve({mockResult: true});
+                cache.captureTask('mockTask', promise);
+
+                // WHEN a task is retrieved
+                const taskPromise = cache.getTaskPromise('mockTask');
+
+                // THEN it should resolve with the same result as the captured task
+                return taskPromise.then((result) => {
+                    expect(result).toEqual({mockResult: true});
+                });
+            });
+        });
+    });
+
+    describe('Onyx with Cache', () => {
+        let Onyx;
+        let withOnyx;
+
+        /** @type OnyxCache */
+        let cache;
+
+        const ONYX_KEYS = {
+            TEST_KEY: 'test',
+            ANOTHER_TEST: 'anotherTest',
+            COLLECTION: {
+                MOCK_COLLECTION: 'mock_collection_',
+            },
+        };
+
+        function initOnyx() {
+            const OnyxModule = require('../../index');
+            Onyx = OnyxModule.default;
+            withOnyx = OnyxModule.withOnyx;
+            cache = require('../../lib/OnyxCache').default;
+
+            Onyx.init({
+                keys: ONYX_KEYS,
+                registerStorageEventListener: jest.fn(),
+            });
+
+            // Onyx init introduces some side effects e.g. calls the getAllKeys
+            // We're clearing mocks to have a fresh calls history
+            return waitForPromisesToResolve()
+                .then(() => jest.clearAllMocks());
+        }
+
+        // Always use a "fresh" instance
+        beforeEach(() => {
+            jest.resetModules();
+            return initOnyx();
+        });
+
+        it('Expect a single call to getItem when multiple components use the same key', () => {
+            const AsyncStorageMock = require('@react-native-community/async-storage/jest/async-storage-mock');
+
+            // GIVEN a component connected to Onyx
+            const TestComponentWithOnyx = withOnyx({
+                text: {
+                    key: ONYX_KEYS.TEST_KEY,
+                },
+            })(ViewWithText);
+
+            // GIVEN some string value for that key exists in storage
+            AsyncStorageMock.getItem.mockResolvedValue('"mockValue"');
```

Comment:
Is there some significance to doing a single and double quote here e.g. `'"mockValue"'` vs. just `'mockValue'`?